src/EditorFeatures/Core/Implementation/CommentSelection/CommentSelectionResult.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection
+{
+    internal struct CommentSelectionResult
+    {
+        /// <summary>
+        /// Text changes to make for this operation.
+        /// </summary>
+        public ImmutableList<TextChange> TextChanges { get; }
+        /// <summary>
+        /// Tracking spans used to format and set the output selection after edits.
+        /// </summary>
+        public ImmutableList<CommentTrackingSpan> TrackingSpans { get; }
+        /// <summary>
+        /// The type of text changes being made.
+        /// </summary>
+        public Operation ResultOperation { get; }
```

Comment:
do you need this here?  don't you know the op you're performing up front?  it seems... odd as a result as it's not really dynamic.

Reply:
For comment/uncomment we know the operation upfront.  For toggle, we are not aware upfront.

Reply:
ah, i see.  this might be worth calling out.